localstack/utils/persistence.py

```diff
@@ -39,7 +39,7 @@ def record(api, method, path, data, headers):
     try:
         if isinstance(data, dict):
             data = json.dumps(data)
-        if data:
+        if data or data in [u'', b'']:
```

Comment:
Perhaps a safer / more general solution would be:
```
import six
...
if isinstance(data, (six.string_types, bytes)):
    ...
```

Reply:
Is not about it. The `if` block should run only if `data` is a "basestring", but the original code runs only if `data` evaluates to True. It leads to ignore empty strings.

You are right in that the "right" code should be alike:
```python
if isinstance(data, basestring):
    ...
```
But it does seems so overkill to import six for it...

Reply:
Oops: Cannot be `isinstance(data, ...)` because will do stuff if `data` is True but is not a string:

https://github.com/alanjds/localstack/blob/1fcbaa3215a24c1d199a5f8056fcc22599cc592f/localstack/utils/persistence.py#L42-L47

Reply:
Not sure I understand. If `data` is `True`, then `to_bytes(data)` and `base64.b64encode(data)` would likely fail.

Note that I'm not opposed to the implementation we have in the current PR. Just trying to understand what the different corner cases could be. But I think for now we should be good with the current implementation.

Reply:
Thanks for the merge.

Seems to me that it converts whatever `to_bytes`. However, I hit a corner case when passing no data from console. It came as "unicode" (or str) and got not converted to bytes. Then failed on serialization.

Reply:
Even empty unicode (or str) should be converted `to_bytes`.